imap-codec/src/fragmentizer.rs

```diff
@@ -563,6 +592,8 @@ pub enum DecodeMessageError<'a, C: Decoder> {
     },
     /// Max message size was exceeded and bytes were dropped.
     MessageTooLong { initial: Secret<&'a [u8]> },
+    /// The message was explicitly poisoned to prevent decoding.
+    MessagePoisoned { undecoded: Secret<&'a [u8]> },
```

Comment:
`undecoded` sounds a bit weird. `raw`? `bytes`? `message_bytes`?

Reply:
The other variants have descriptive names for the bytes. I like `discarded`.